R/roles.R

```diff
@@ -217,6 +217,24 @@ add_role <- function(recipe, ..., new_role = "predictor", new_type = NULL) {
     recipe$var_info$source[first_row_with_var]
   })
 
+  for (var in vars) {
+    old_roles <- recipe$var_info$role[recipe$var_info$variable == var]
+
+    if (new_role == "predictor" && any(old_roles == "outcome")) {
+      cli::cli_abort(
+        "{.var {var}} cannot get {.val predictor} role as it is already \\
+          has role {.val outcome}."
+        )
+    }
+    
+    if (new_role == "outcome" && any(old_roles == "predictor")) {
+      cli::cli_abort(
+        "{.var {var}} cannot get {.val outcome} role as it is already \\
+        has role {.val predictor}."
+      )
+    }
+  }
+
```

Comment:
it felt cleaner to have 2 `cli::cli_abort()` calls than trying to do something clever to avoid it.

Also all the `"case_weights"` is handled elsewhere in the code, so no need to check for that here

Reply:
Agree with erroring here 👍